curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/sorted-union.md

```diff
@@ -62,7 +74,7 @@ uniteUnique([1, 3, 2], [5, 2, 1, 4], [2, 1]);
 ```js
 function uniteUnique(arr) {
   return [].slice.call(arguments).reduce(function(a, b) {
-    return [].concat(a, b.filter(function(e) {return a.indexOf(e) === -1;}));
+    return [].concat(a, [...new Set(b)].filter(function(e) {return a.indexOf(e) === -1;}));
```

Comment:
Thank you for the update! This approach works as expected.

I'm sorry to be a pain, but since there is a Functional Programming section right before this one that helps the users get familiar with array methods, I wonder if we should apply those in the solution instead of introducing a new data type.

My suggestion is to add another condition to the filtering of `b`:

```suggestion
    return [].concat(
      a, 
      b.filter(function(e, currentIndex) {
        return b.indexOf(e) === currentIndex && a.indexOf(e) === -1;
      }));
```

With `indexOf` returning the index of the first element it can find, the `b.indexOf(e) === currentIndex` check will fail when it is run against duplicate items, and leave them out of the final array.

Reply:
Thanks again for the suggestion. I understand that these courses are supposed to be beginner-friendly and we have to take a lot of things into consideration while writing the solutions.

Since you graciously provided the exact code snippet I needed to use, I have added it to the solution instead of my previous approach.

Would greatly appreciate it if you could take another look at my code and see if it's now looking and working as expected.